odoo/addons/base/models/ir_attachment.py

```diff
@@ -343,7 +343,9 @@ def _check_contents(self, values):
         xml_like = 'ht' in mimetype or ( # hta, html, xhtml, etc.
                 'xml' in mimetype and    # other xml (svg, text/xml, etc)
                 not 'openxmlformats' in mimetype)  # exception for Office formats
-        user = self.env.context.get('binary_field_real_user', self.env.user)
+        user = self.env.context.get('binary_field_real_user')
+        if not isinstance(user, self.pool['res.users']):
+            user = self.env.user
```

Comment:
Why not simply do something like: 
```suggestion
        user = self.env.context.get('binary_field_real_user', self.env.user).id
```

Reply:
This smart, but perhaps a bit too smart. This doesn't really convey the intention which is to ensure that the content of `binary_field_real_user` is a record. One could refactor the code and miss it, inadvertently removing the check.

Reply:
at least raise error instead of falling back to current user?

Reply:
> at least raise error instead of falling back to current user?

Changes done 👌🏻